### PR TITLE
Added support for Autotuning API

### PR DESCRIPTION
**Pull request by kosta-bizetic**
## Description

---

I generated C# libraries using Autorest for these two APIs:

[https://docs.microsoft.com/en-us/rest/api/sql/serverautomatictuning](url)
[https://docs.microsoft.com/en-us/rest/api/sql/databaseautomatictuning](url)

and added a test for each API

Swagger specs were already in azure-rest-api-specs. I added them to the corresponding readme.md.

Link to the azure-rest-api-specs PR [https://github.com/Azure/azure-rest-api-specs/pull/2325]()

This checklist is used to make sure that common guidelines for a pull request are followed.
- [x] Please add REST spec PR link to the SDK PR
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Use generate.cmd to generate you SDK
Second build your project as below
msbuild build.proj /t:build /p:Scope=SDKs\SqlManagement
this will create a meta data file that is required (if there are any changes in the API version)

**Comment by shahabhijeet** [comment]
@kosta-bizetic get onboarded to github by visting aka.ms/azuregithub
This will allow you to get required permisison to kick off CI process.
Also make sure you create an account in travis by visiting travis-ci.org

**Comment by kosta-bizetic** [comment]
I have created an account on travis-ci.org so I should have the permissions now.

I have built the SDK prior to this PR and even when I build it now using your command I still don't see any meta data file. Where should it appear?

**Comment by kosta-bizetic** [comment]
Here is the link to the new PR for azure-rest-api-specs [https://github.com/Azure/azure-rest-api-specs/pull/2332](https://github.com/Azure/azure-rest-api-specs/pull/2332)

File: src/SDKs/SqlManagement/Sql.Tests/AutomaticTuningOperationsTests.cs [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                ResourceGroup resourceGroup = context.CreateResourceGroup();
+                Server server = context.CreateServer(resourceGroup);
+                SqlManagementClient sqlClient = context.GetClient<SqlManagementClient>();
+
+                // Set server autotuning options Custom
+                ServerAutomaticTuning parameters = new ServerAutomaticTuning() {
+                    DesiredState = AutomaticTuningServerMode.Custom,
+                    Options = new Dictionary<string, AutomaticTuningServerOptions>()
+                    {
+                        {"createIndex", new AutomaticTuningServerOptions() {DesiredState=AutomaticTuningOptionModeDesired.On} },
```
Comment (jaredmoo) [inline src/SDKs/SqlManagement/Sql.Tests/AutomaticTuningOperationsTests.cs]: is there an enum for these options?

**Review by jaredmoo — APPROVED** [review]

File: src/SDKs/SqlManagement/Sql.Tests/AutomaticTuningOperationsTests.cs (hunk 1 above)
Comment (kosta-bizetic) [reply src/SDKs/SqlManagement/Sql.Tests/AutomaticTuningOperationsTests.cs]: There isn't

**Comment by kosta-bizetic** [comment]
@shahabhijeet can you tell me more about the meta data file?

**Comment by shahabhijeet** [comment]
@azuresdkci retest this please

**Comment by shahabhijeet** [comment]
@azuresdkci test this please

**Comment by kosta-bizetic** [comment]
@shahabhijeet 
I merged changes from psSdkJson6 into my branch, regenerated and rebuilt the code, and run the tests again. Is the sql_resource-manager.txt from the latest commit the meta data file that should appear?

**Comment by shahabhijeet** [comment]
@kosta-bizetic I do not see Jenkins job completed for your PR.
Unless Jenkins job is a success, this PR cannot be merged.
Have you onboarded using aka.ms/azuregithub?

**Comment by kosta-bizetic** [comment]
@shahabhijeet All checks have passed. Can the PR be merged now?

**Comment by kosta-bizetic** [comment]
@shahabhijeet All checks have passed. Can the PR be merged now?

**Review by shahabhijeet — APPROVED** [review]

**Pull request merged**